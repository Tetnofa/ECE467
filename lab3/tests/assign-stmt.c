void f2(int n) {
  
}

void f1() {
  int a;
  int b;
  bool c;
  bool d;
  a = b;
  f2(a);
  a = a == a;
  a = c;
  c = a;
  c = a + c;
  f2(a);
}
