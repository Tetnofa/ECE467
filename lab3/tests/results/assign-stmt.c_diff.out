0a1,71
> Program [useIO=0](1,0)
> 	Function[isProto=0]	Void(1,0)
> 		Identifier[name:f2](1,5)
> 		Parameter		Int(1,8)
> 			Identifier[name:n](1,12)
> 		Scope(1,15)
> 	Function[isProto=0]	Void(5,0)
> 		Identifier[name:f1](5,5)
> 		Scope(5,10)
> 			Scalar Declaration			Int(6,2)
> 				Identifier[name:a](6,6)
> 			Scalar Declaration			Int(7,2)
> 				Identifier[name:b](7,6)
> 			Scalar Declaration			Bool(8,2)
> 				Identifier[name:c](8,7)
> 			Scalar Declaration			Bool(9,2)
> 				Identifier[name:d](9,7)
> 			Assignment(10,2)
> 				Reference(10,2)
> 					Identifier[name:a](10,2)
> 				Int Expression(10,6)
> 					Reference(10,6)
> 						Identifier[name:b](10,6)
> 			Expression Statement(11,2)
> 				CallExpr(11,2)
> 					Identifier[name:f2](11,2)
> 					Argument(11,5)
> 						Int Expression(11,5)
> 							Reference(11,5)
> 								Identifier[name:a](11,5)
> 			Assignment(12,2)
> 				Reference(12,2)
> 					Identifier[name:a](12,2)
> 				Bool Expression(12,6)
> 					Binary Expression[opcode:==](12,6)
> 						Int Expression(12,6)
> 							Reference(12,6)
> 								Identifier[name:a](12,6)
> 						Int Expression(12,11)
> 							Reference(12,11)
> 								Identifier[name:a](12,11)
> 			Assignment(13,2)
> 				Reference(13,2)
> 					Identifier[name:a](13,2)
> 				Int Expression(13,6)
> 					Reference(13,6)
> 						Identifier[name:c](13,6)
> 			Assignment(14,2)
> 				Reference(14,2)
> 					Identifier[name:c](14,2)
> 				Int Expression(14,6)
> 					Reference(14,6)
> 						Identifier[name:a](14,6)
> 			Assignment(15,2)
> 				Reference(15,2)
> 					Identifier[name:c](15,2)
> 				Int Expression(15,6)
> 					Binary Expression[opcode:+](15,6)
> 						Int Expression(15,6)
> 							Reference(15,6)
> 								Identifier[name:a](15,6)
> 						Int Expression(15,10)
> 							Reference(15,10)
> 								Identifier[name:c](15,10)
> 			Expression Statement(16,2)
> 				CallExpr(16,2)
> 					Identifier[name:f2](16,2)
> 					Argument(16,5)
> 						Int Expression(16,5)
> 							Reference(16,5)
> 								Identifier[name:a](16,5)
